src/ethereum/genesis.py

```diff
@@ -143,28 +148,52 @@ def add_genesis_block(
     genesis :
         The genesis configuration to use.
     """
-    for account, balance in genesis.initial_balances.items():
-        hardfork.state.create_ether(chain.state, account, balance)
-
-    genesis_header = hardfork.eth_types.Header(
-        parent_hash=hardfork.eth_types.Hash32(b"\0" * 32),
-        ommers_hash=rlp.rlp_hash(()),
-        coinbase=Address(b"\0" * 20),
-        state_root=hardfork.state.state_root(chain.state),
-        transactions_root=hardfork.trie.root(hardfork.trie.Trie(False, None)),
-        receipt_root=hardfork.trie.root(hardfork.trie.Trie(False, None)),
-        bloom=hardfork.eth_types.Bloom(b"\0" * 256),
-        difficulty=genesis.difficulty,
-        number=Uint(0),
-        gas_limit=genesis.gas_limit,
-        gas_used=Uint(0),
-        timestamp=genesis.timestamp,
-        extra_data=genesis.extra_data,
-        mix_digest=hardfork.eth_types.Hash32(b"\0" * 32),
-        nonce=genesis.nonce,
-    )
+    for address, account in genesis.initial_accounts.items():
+        address = hardfork.utils.hexadecimal.hex_to_address(address)
```

Comment:
What's the advantage of storing a string in `initial_accounts` over doing the conversion in the constructor?